components/github_alert.lua

```diff
@@ -0,0 +1,89 @@
+local util = require 'util'
+local json = require 'json'
+local http_request = require 'http.request'
+local os = require 'os'
+local io = require 'io'
+local luadate = require 'date'
+commit = false
+commit_of_day = 0
+first_time = true
+
+require 'cairo'
+
+-- an component to remind your github little green dot :P
+local gh_endpoint = "https://api.github.com"
```

Comment:
If a variable is only used in and for a single function, put it in that function is recommended.